#! /bin/sh /usr/share/dpatch/dpatch-run
## 02unicode.dpatch by  <az@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: fix unicode decode-does-encode-and-fail problems with iso8859 locales

@DPATCH@
diff -urNad '--exclude=CVS' '--exclude=.svn' '--exclude=.git' '--exclude=.arch' '--exclude=.hg' '--exclude=_darcs' '--exclude=.bzr' duplicity-0.6.20~/bin/duplicity duplicity-0.6.20/bin/duplicity
--- duplicity-0.6.20~/bin/duplicity	2013-05-22 12:21:37.000000000 +1000
+++ duplicity-0.6.20/bin/duplicity	2013-05-22 12:21:37.000000000 +1000
@@ -30,6 +30,10 @@
 import getpass, gzip, os, sys, time, types
 import traceback, platform, statvfs, resource, re
 
+# override locale to avoid bug #682837, until
+# the logger finally deals with locales cleanly
+os.environ['LC_ALL']="POSIX"
+
 pwd = os.path.abspath(os.path.dirname(sys.argv[0]))
 if os.path.exists(os.path.join(pwd, "../duplicity")):
     sys.path.insert(0, os.path.abspath(os.path.join(pwd, "../.")))
